pkg/test/framework/components/istio/operator.go

```diff
@@ -693,3 +708,132 @@ func deployCACerts(workDir string, env *kube.Environment, cfg Config) error {
+			Name:      istiodSvcName,
+			Namespace: cfg.SystemNamespace,
+		},
+		Spec: kubeApiCore.ServiceSpec{
+			Ports: []kubeApiCore.ServicePort{
+				{
+					Port:     15012,
+					Name:     "tcp-istiod",
+					Protocol: kubeApiCore.ProtocolTCP,
+				},
+				{
+					Name:     "tcp-webhook",
+					Protocol: kubeApiCore.ProtocolTCP,
+					Port:     15017,
+				},
+			},
+			ClusterIP: "None",
+		},
+	}
+	_, err := cluster.CoreV1().Services(cfg.SystemNamespace).Create(context.TODO(), svc, kubeApiMeta.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	eps := &kubeApiCore.Endpoints{
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name:      istiodSvcName,
+			Namespace: cfg.SystemNamespace,
+		},
+		Subsets: []kubeApiCore.EndpointSubset{
+			{
+				Addresses: []kubeApiCore.EndpointAddress{
+					{
+						IP: discoveryAddress,
+					},
+				},
+				Ports: []kubeApiCore.EndpointPort{
+					{
+						Name:     "tcp-istiod",
+						Protocol: kubeApiCore.ProtocolTCP,
+						Port:     15012,
+					},
+					{
+						Name:     "tcp-webhook",
+						Protocol: kubeApiCore.ProtocolTCP,
+						Port:     15017,
+					},
+				},
+			},
+		},
+	}
+
+	_, err = cluster.CoreV1().Endpoints(cfg.SystemNamespace).Create(context.TODO(), eps, kubeApiMeta.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	err = retry.UntilSuccess(func() error {
+		_, err := cluster.CoreV1().Services(cfg.SystemNamespace).Get(context.TODO(), istiodSvcName, kubeApiMeta.GetOptions{})
+		if err != nil {
+			return err
+		}
+		_, err = cluster.CoreV1().Endpoints(cfg.SystemNamespace).Get(context.TODO(), istiodSvcName, kubeApiMeta.GetOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	}, retry.Timeout(20*time.Second), retry.Delay(2*time.Second))
+	return err
+}
+
+func deployWithRemoteKubeConfig(iopFile, remoteIopFile string, env *kube.Environment, cfg Config, i *operatorComponent) error {
+	var err error
+	// 1. install istiod Remote + base in remote cluster
+	for _, cluster := range env.KubeClusters {
+		if !env.IsControlPlaneCluster(cluster) {
+			cluster := cluster
+			if err := deployControlPlane(i, cfg, cluster, remoteIopFile); err != nil {
+				return fmt.Errorf("failed deploying control plane to cluster %s: %v", cluster.Name(), err)
+			}
+			break
+		}
+	}
+
+	var istiodAddress net.TCPAddr
+	for _, cluster := range env.KubeClusters {
+		if env.IsControlPlaneCluster(cluster) {
+			// 2. install istiod on primary cluster
+			if err := deployControlPlane(i, cfg, cluster, iopFile); err != nil {
+				return fmt.Errorf("failed deploying control plane to cluster %s: %v", cluster.Name(), err)
+			}
+			// 3. get Istiod address as discovery Address
+			err = retry.UntilSuccess(func() error {
+				istiodAddress, err = i.RemoteDiscoveryAddressFor(cluster)
+				if err != nil {
+					return err
+				}
+				return nil
+			}, retry.Timeout(90*time.Second), retry.Delay(2*time.Second))
+			if err != nil {
+				return err
+			}
+			err = patchIstiodCustomHost(i, cfg, cluster)
```

Comment:
Is this changing the host on the primary cluster? That seems wrong.

Reply:
this is to add istiod's IP address into ISTIOD_CUSTOM_HOST in the primary cluster.

Reply:
Is there any reason to not just always do this and just make it part of the default install of a primary cluster?